### PR TITLE
Add callbacks and promises exercises

### PR DESCRIPTION
**Pull request by 40thieves**
Adds exercises for callbacks and promises.

I mostly just adapted from them old syllabus, which had them in this [CodePen](https://codepen.io/fortythieves/pen/QOYgWb?editors=1010).

@BexB1 @kkarimi @kabaros

### CODE REVIEW

File: week-5/B-promises/exercise.js [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+
+function exercise1() {
+  // Write your implementation here
+}
+
+/*
+  EXERCISE 2
+  =======
+
+  Write a callback function that inserts the result into the #exercise2
```
Comment (djgrant) [inline week-5/B-promises/exercise.js]: I think some documentation is needed to explain how this callback gets called.

Comment (djgrant) [reply week-5/B-promises/exercise.js]: Wasn't clear to me until I scrolled down and looked at the function definition for functionThatCallsBack

File: week-5/B-promises/exercise.js [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+*/
+functionThatCallsBack(function() {
+  // Write your implementation here
+})
+
+/*
+  EXERCISE 3
+  =======
+
+  Write a function that calls the callback with "Hello from the callback"
```
Comment (djgrant) [inline week-5/B-promises/exercise.js]: ....with _the parameter_ "Hello from the callback"

**Comment by 40thieves** [comment]
Thanks for the review 😄

> I think it would be good for them to be tackled in separate files.

Yeah, I was thinking that week5 has a lot of stuff now. But as I mentioned on slack I think we might want to split up this content a bit more (which we should do separately). I've moved the callbacks exercises to their own directory.

I've also addressed your other feedback.

**Comment by djgrant** [comment]
Looks good!

**Comment by 40thieves** [comment]
Cheers! Merging

**Pull request merged**